client/src/components/ChannelFilter.tsx

```diff
@@ -0,0 +1,81 @@
+import Select, {SingleValue} from 'react-select';
+import {ChannelFilterProps} from '../lib/types';
+import React from 'react';
+import styled from 'styled-components';
+import {useCallback, useEffect, useState} from 'react';
+import {fetchData} from '../services/fetch';
+
+interface Channel {
+    id: string;
+    name: string;
+}
+
+interface Option {
+    label: string;
+    value: string;
+}
+
+const FilterDiv = styled.div`
+      min-width: 350px;
+      max-width: 50%;
+      flex-grow: 1;
+    `;
+
+const ChannelFilter = ({onSelected, selectedChannel}: ChannelFilterProps) => {
+    const allValue = {
+        label: 'Code Reviews For All Slack Channels',
+        value: 'all',
+    };
+    const [channelOptions, setChannelOptions] = useState([] as Option[]);
+    const [selectPlaceHolder, setSelectPlaceHolder] = useState(allValue.label);
+    const handleChannelSelectionChange = useCallback((data: SingleValue<Option>) => {
+        if (data?.label && data?.value) {
+            setSelectPlaceHolder(data.label);
+            onSelected(data.value);
+        }
+    }, [onSelected]);
+
+    useEffect(() => {
+        fetchData('/api/channels')
+            .then((result) => {
+                const options: Option[] = [
+                    allValue,
+                    ...result.map((channel: Channel) => {
+                        return {
+                            label: `Code Reviews For Channel "#${channel.name}"`,
+                            value: channel.id,
+                        }
+                    }),
+                ];
+
+                setChannelOptions(options);
+
+                // This code is to work around to display current selected channel as a placeholder
+                // @TODO: revisit on how to pass in the default value to <Select> after the options are set.
+                if (selectedChannel === 'all') {
+                  return;
+                }
+                
+                const findChannel = result.find((channel: Channel) => [channel.id, channel.name].includes(selectedChannel))
+                if (findChannel) {
+                    setSelectPlaceHolder(`Code Reviews For Channel "#${findChannel.name}"`);
+                    if (selectedChannel !== findChannel.id) {
+                        onSelected && onSelected(findChannel.id);
```

Comment:
Is this `&&` operator just to ensure the function is not undefined before calling it? Why not just do

```js
onSelected(findChannel.id);
```

Reply:
Good catch, copy & paste from existing code.  This field always exists.